src/routes/cache-hooks/cache-hooks.controller.ts

```diff
@@ -43,6 +43,6 @@ export class CacheHooksController {
       | PendingTransaction
       | SafeAppsUpdate,
   ): Promise<void> {
-    await this.service.onEvent(eventPayload);
+    this.service.onEvent(eventPayload);
```

Comment:
This is changed to run the `onEvent` callback in a non-blocking fashion and avoid having the caller waiting for the event hook execution.

Reply:
I think it'd be simpler if the delay was tackled on a web-hook route only initially. So you'd have something like

```typescript
setTimeout(() => {
  this.service.onEvent(eventPayload);
}, 5000); // read timeout value from the configuration
```

I would also add a small test to this where the `service` is mocked and we use virtual timers. With the mock you can check that the service was only called after T seconds/milliseconds. Due to the scope of this issue it's also fine if the test is added as a separate PR 👍

Reply:
I've changed it in [563547d](https://github.com/safe-global/safe-client-gateway/pull/883/commits/563547d36d6c47eb6cdbbab2242173d5723f4fbf)

I had to adjust the controller tests in this PR, they would fail otherwise, as `fakeCacheService.get` wouldn't return `null` immediately as the controller tests were expecting.

Reply:
I also changed the status code returned by the controller to `202` as it is more accurate given the new behavior.